projects/plugins/jetpack/extensions/blocks/ai-paragraph/edit.js

```diff
@@ -4,222 +4,263 @@ import apiFetch from '@wordpress/api-fetch';
-	}
+	const tagObjects = useSelect( select => {
+		return tags
+			.map( tagId => {
+				const tag = select( 'core' ).getEntityRecord( 'taxonomy', 'post_tag', tagId );
 
-	// We are grabbing more data from WP.
-	const categoryObjects = categories.map( categoryId =>
-		select( 'core' ).getEntityRecord( 'taxonomy', 'category', categoryId )
-	);
-	const tagObjects = tags.map( tagId =>
-		select( 'core' ).getEntityRecord( 'taxonomy', 'post_tag', tagId )
-	);
+				if ( ! tag ) {
+					// Data is not yet loaded
+					loading = true;
+					return;
+				}
 
-	// For now categories are combined with tags into the same object. I am not sure they should
-	const taxonomies = categoryObjects.concat( tagObjects );
+				return tag;
+			} )
+			.filter( Boolean ); // Remove undefined values
+	} );
 
-	// We want to wait until all category names are loaded. This will return empty string (aka loading state) until all objects are truthy.
-	if ( taxonomies.filter( obj => ! obj || ! obj.name ).length ) {
-		return '';
+	if ( loadingCategories !== loading ) {
+		setLoadingCategories( loading );
 	}
 
+	const createPrompt = ( title = '', content = '', categoryNames = '' ) => {
+		content = content.slice( -240 );
+
+		// If title is not added, we will only complete.
+		if ( ! title ) {
+			return content;
+		}
+
+		// Title, content and categories
+		if ( content && categoryNames.length ) {
+			return sprintf(
+				/** translators: This will be a prompt to OpenAI to generate a post based on the post title, comma-seperated category names and the last 240 characters of content. */
+				__( "This is a post titled '%1$s', published in categories '%2$s':\n\n … %3$s", 'jetpack' ), // eslint-disable-line @wordpress/i18n-no-collapsible-whitespace
+				title,
+				categoryNames,
+				content
+			);
+		}
+
+		// No content, only title and categories
+		if ( categoryNames.length ) {
+			return sprintf(
+				/** translators: This will be a prompt to OpenAI to generate a post based on the post title, and comma-seperated category names. */
+				__( "This is a post titled '%1$s', published in categories '%2$s':\n", 'jetpack' ), // eslint-disable-line @wordpress/i18n-no-collapsible-whitespace
+				title,
+				categoryNames
+			);
+		}
+
+		// Title and content
+		if ( content ) {
+			return sprintf(
+				/** translators: This will be a prompt to OpenAI to generate a post based on the post title, and the last 240 characters of content. */
+				__( "This is a post titled '%1$s':\n\n…%2$s", 'jetpack' ), // eslint-disable-line @wordpress/i18n-no-collapsible-whitespace
+				title,
+				content
+			);
+		}
+
+		return sprintf(
+			/** translators: This will be a prompt to OpenAI to generate a post based on the post title */
+			__( 'Write content of a post titled "%s"', 'jetpack' ),
+			title
+		);
+	};
+
+	const containsAiUntriggeredParapgraph = _content =>
+		_content.filter(
+			block => block.name && block.name === 'jetpack/ai-paragraph' && ! block.attributes.content
+		).length > 0;
```

Comment:
I know this works, but I would feel better if there was `{` and a return statement...

Reply:
Totally valid remark :)

Reply:
Done in f3b74104f8cd5dcbc480dfb9eacd381923fe7dd1